addons/delivery/models/sale_order.py

```diff
@@ -38,6 +38,7 @@ def onchange_order_line(self):
 
     def _remove_delivery_line(self):
         self.env['sale.order.line'].search([('order_id', 'in', self.ids), ('is_delivery', '=', True)]).unlink()
+        self.recompute_delivery_price = True
```

Comment:
As `recompute_delivery_price` is never set back to `False`, once this code run, user won't be able to ever go through checkout no?